akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -54,13 +54,15 @@ class LightArrayRevolverScheduler(
 
   import LightArrayRevolverScheduler._
 
-  private val oneNs = Duration.fromNanos(1l)
-  private def roundUp(d: FiniteDuration): FiniteDuration =
-    try {
-      ((d + TickDuration - oneNs) / TickDuration).toLong * TickDuration
-    } catch {
-      case _: IllegalArgumentException ⇒ d // rounding up Long.MaxValue.nanos overflows
+  private def roundUp(d: FiniteDuration): FiniteDuration = {
+    val dn = d.toNanos
+    if (dn <= 0)
+      Duration.Zero
+    else {
+      val r = ((dn - 1) / tickNanos + 1) * tickNanos
+      if (r != dn && r > 0) r.nanos else d
```

Comment:
I think @rkuhn suggested that the `Duration.Zero` if branch can be removed because of the `r >0` check here.

I don't know if negative is alright? Old code didn't change negative duration to Zero.